### PR TITLE
Swift lint and fastlane

### PR DESCRIPTION
**Pull request by jankrzempek**
From now, we are going to use the AirCasting date style, which will be defined in the DataBuilder.
FastLane has a new lane that can check any occurrences of Date() in our code and leftover prints.

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    capture_group: 0 # number of regex capture group to highlight the rule violation at. optional.
+    match_kinds: # SyntaxKinds to match. optional.
+      - identifier
+    message: "You! Use Air Casting date format - use Date().currentUTCTimeZoneDate" # violation message. optional.
+    severity: error # violation severity. optional
+  
+  print_using:
+    included: ".*\\.swift" # regex that defines paths to include during linting. optional.
+    excluded: ".*Tests\\.swift" # regex that defines paths to exclude during linting. optional
+    regex: 'print'
```
Comment (pawelgil) [inline .swiftlint.yml:98]: well this regex will produce some false positives like:

`printDesctiption()`

File: AirCasting/Extensions/DataBuilder.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+class DateBuilder {
+    static func getDate() -> Date {
+        Date().currentUTCTimeZoneDate
+    }
+    
+    static func getRawDate() -> Date {
+        Date()
+    }
+    
+    static func getSince1970using(_ timeInterval: Double) -> Date {
```
Comment (pawelgil) [inline AirCasting/Extensions/DataBuilder.swift:15]: `getDateWithTimeIntervalSince1970(_ timeInterval: Double) -> Date`

File: AirCasting/Extensions/DataBuilder.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        Date(timeIntervalSince1970: timeInterval)
+    }
+    
+    static func getSince1970() -> Double {
```
Comment (pawelgil) [inline AirCasting/Extensions/DataBuilder.swift:19]: `getTimeIntervalSince1970() -> Double`

File: AirCasting/Extensions/DataBuilder.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        Date().timeIntervalSince1970
+    }
+    
+    static func getSince(timeInterval: Double) -> Date {
```
Comment (pawelgil) [inline AirCasting/Extensions/DataBuilder.swift:23]: `getDateWithTimeIntervalSinceReferenceDate(_ timeInterval: Double)  -> Date`

File: AirCasting/Extensions/DataBuilder.swift [hunk 5]
```diff
@@ -0,0 +1,39 @@
+// Created by Lunar on 24/01/2022.
+//
+import Foundation
+
+// swiftlint:disable airCasting_date
+class DateBuilder {
+    static func getDate() -> Date {
```
Comment (pawelgil) [inline AirCasting/Extensions/DataBuilder.swift:7]: let's be explicit here and use appropriate names: `getDateMovedToUTCZone ` or `getFakeUTCDate`, idk. Something better. Also please add documentation for each function that is not self explanatory (add a comment with tripple slash `///` above function)

File: AirCasting/Extensions/Date+Miliseconds.swift [hunk 6]
```diff
@@ -7,7 +7,7 @@ extension Date {
     static let msFormatter: DateFormatter = DateFormatters.DateExtension.milisecondsDateFormatter
     
     var milliseconds: Int {
```
Comment (pawelgil) [inline AirCasting/Extensions/Date+Miliseconds.swift:9]: please remove changes from this extension and add `swiftlint:disable` marks here too

File: AirCasting/Graph/StatisticsContainer/Adapters/GraphStatsDataSource.swift [hunk 7]
```diff
@@ -2,6 +2,7 @@
 //
 
 import Foundation
+import Gzip
```
Comment (pawelgil) [inline AirCasting/Graph/StatisticsContainer/Adapters/GraphStatsDataSource.swift:5]: don't think that's needed here

File: Podfile.lock [hunk 8]
```diff
@@ -139,6 +140,7 @@ DEPENDENCIES:
   - GoogleMaps (= 4.1.0)
   - GooglePlaces (= 4.1.0)
   - GzipSwift
+  - SwiftLint
```
Comment (pawelgil) [inline Podfile.lock:143]: the community is slowly moving away from Pods, is it possible to add SwiftLint using SPM?

**Review by pawelgil — CHANGES_REQUESTED** [review]

File: .swiftlint.yml [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+custom_rules:
+  airCasting_date: # rule identifier
+    included: ".*\\.swift" # regex that defines paths to include during linting. optional.
+    excluded: ".*Tests\\.swift" # regex that defines paths to exclude during linting. optional
+    name: "Air Casting date" # rule name. optional.
+    regex: "(\\W|^)Date(\\(|\\.)" # matching pattern
+    capture_group: 0 # number of regex capture group to highlight the rule violation at. optional.
+    match_kinds: # SyntaxKinds to match. optional.
+      - identifier
+    message: "You! Use Air Casting date format - use Date().currentUTCTimeZoneDate" # violation message. optional.
```
[the remark this answers is not in the document]
Comment (jankrzempek) [reply .swiftlint.yml:92]: 😬

File: AirCasting/Graph/StatisticsContainer/Adapters/GraphStatsDataSource.swift (hunk 7 above)
Comment (jankrzempek) [reply AirCasting/Graph/StatisticsContainer/Adapters/GraphStatsDataSource.swift:5]: It was done by mistake

File: Podfile.lock (hunk 8 above)
Comment (jankrzempek) [reply Podfile.lock:143]: I checked and for my knowledge it is not possible, at least for now

**Comment by pawelgil** [comment]
Let's remove swiftlint from Pods and require developers to install it using homebrew:
`brew install swiftlint`
Add information in the README.md plz

Reasons for it:
1) We want to check in Pods/ directory to git and swiftlint will add quite a large number of files there which will bloat our repo
2) This is not a code-level dependency just like fastlane
3) We want to move code-level dependencies to SPM in the future and remove cocoapods completely

**Review by pawelgil — CHANGES_REQUESTED** [review]
Let's remove this from pods as mentioned

**Review by pawelgil — APPROVED** [review]
Splendid!

**Pull request merged**